weave-js/src/components/PagePanelComponents/Home/Browse3/pages/DocumentView/MetadataViewerSection.tsx

```diff
@@ -0,0 +1,242 @@
+import {Collapse} from '@mui/material';
+import {GridRowId, useGridApiRef} from '@mui/x-data-grid-pro';
+import _ from 'lodash';
+import React, {
+  Fragment,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import {Button} from '../../../../../Button';
+import {CodeEditor} from '../../../../../CodeEditor';
+import {ObjectViewer} from '../CallPage/ObjectViewer';
+import {Subheader} from './StyledText';
+
+type MetadataViewerSectionProps = {
+  title: string;
+  data: Data;
+  open?: boolean;
+  error?: string;
+};
+
+const EXPANDED_IDS_LENGTH = 200;
+
+type Data = Record<string, any>;
+
+const MetadataViewerSectionInner = ({
+  title,
+  data,
+  open,
+}: MetadataViewerSectionProps) => {
+  const apiRef = useGridApiRef();
+  // Update this when we change the state to hidden
+  // That way it restores the last state when uncollapsed
+  const [isOpen, setIsOpen] = useState(open ?? false);
+  const [mode, setMode] = useState('collapsed');
```

Comment:
Should we use an enum for this?

Reply:
I updated to an enum in both spots. Resolving